### ISSUE TITLE
Parentheses and other symbols always "highlighted"

### ISSUE THREAD

**Issue by carlosgaldino** [fbergroth/emacs-rustfmt#9]
I added the hook to run `rustfmt` when saving the buffer and sometimes after saving the buffer stays like the image below, with parentheses and other symbols (`<`. `{`, etc) as if they were highlighted (but they're not, I tried to `unhighligit-region` and it didn't work).

Do you have any idea on why this might be happening?

Thank you.

![](http://img.carlosgaldino.com/Emacsmrwhite.local_2016-01-07_08-18-31.png)

**Comment by fbergroth** [comment]
It's likely a minor mode interfering. Which modes do you have enabled (`C-h m`)?

**Comment by carlosgaldino** [comment]
```
Enabled minor modes: Anzu Auto-Composition Auto-Compression Auto-Encryption
Auto-Highlight-Symbol Bug-Reference-Prog Clean-Aindent Column-Number Company
Electric-Indent Eval-Sexp-Fu-Flash Evil Evil-Escape Evil-Jumper Evil-Local
Evil-Search-Highlight-Persist Evil-Surround File-Name-Shadow Flx-Ido Flycheck
Flycheck-Pos-Tip Font-Lock Global-Anzu Global-Auto-Revert
Global-Evil-Search-Highlight-Persist Global-Evil-Surround Global-Font-Lock
Global-Hl-Line Global-Linum Global-Page-Break-Lines
Global-Spacemacs-Default-Map-Override Global-Spacemacs-Leader-Override
Global-Undo-Tree Global-Vi-Tilde-Fringe Goto-Address-Prog Helm Helm-Descbinds
Helm-Flx Highlight-Numbers Highlight-Parentheses Hl-Todo Hs Ido-Vertical
Info-Breadcrumbs-In-Mode-Line Line-Number Linum Mac-Mouse-Wheel Menu-Bar
Override-Global Persp Popwin Projectile Projectile-Global Rainbow-Delimiters
Recentf Savehist Shell-Dirtrack Show-Smartparens Show-Smartparens-Global
Smartparens Spaceline-Helm Spaceline-Info Spacemacs-Default-Map-Override
Spacemacs-Leader-Override Transient-Mark Undo-Tree Vi-Tilde-Fringe
Volatile-Highlights Which-Key Window-Numbering Winner Xterm-Mouse Yas Yas-Global
```

:cold_sweat:

**Comment by fbergroth** [comment]
Thanks, unfortunately I can't reproduce it. :disappointed: 

I'm also using spacemacs, so I have basically the same setup. Try updating your packages. If you still see this error, try pinpointing the minor mode by going through them one by one, disable it, and see if the highlighting goes away.

**Comment by carlosgaldino** [comment]
I think it's a problem related to `flycheck-rust`. This error here only appears in a project that flycheck runs and `rustc` returns a non-zero exit code. I tried a couple of times to reproduce in another project where flycheck doesn't give the same error and couldn't reproduce.

I'm waiting on a reply from them.

Btw, since you're also using spacemacs and couldn't reproduce the error, what's your config? Are you using flycheck?

Thanks.

**Comment by fbergroth** [comment]
It would surprise me if the highlighting error was due to flycheck. I'm using flycheck and haven't seen the error you're describing. As a last resort, if you provide me with your `~/.spacemacs`, `SPC h d s` and a sample file which triggers the error, I can try reproducing it.

**Comment by carlosgaldino** [comment]
`~/.spacemacs`: https://github.com/carlosgaldino/dungeon_files/blob/master/dotfiles/spacemacs
Code to trigger the error: https://github.com/carlosgaldino/emacs-error

`SPC h d s`:

``````
#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.4
- Spacemacs branch: master (rev. da94a25)
- Distribution: spacemacs
- Layers:
```elisp
(auto-completion ruby haskell syntax-checking git github javascript c-c++ markdown org html rust shell themes-megapack)
``````

**Comment by fbergroth** [comment]
I ran spacemacs on `da94a25` with your dotfile, and couldn't reproduce the highlighting bug. Also flycheck-cargo works on my end. Maybe it's related to your emacs build.

If you want to debug the rust-cargo checker, you can run `M-x flycheck-compile`, and select `rust-cargo` to see the invocation command and it's output. I'm quite clueless about the highlighting bug though :disappointed:

**Comment by carlosgaldino** [comment]
Yeah, the output is very similar to the output posted on the [initial issue on flycheck-rust](https://github.com/flycheck/flycheck-rust/issues/16) about the error message I get from `rust-cargo`. FYI:

```
cargo rustc -- -Z no-trans --test -L/Users/carlosgaldino/code/sandbox/emacs-error/target/debug -L/Users/carlosgaldino/code/sandbox/emacs-error/target/debug/deps
extra arguments to `rustc` can only be passed to one target, consider filtering
the package by passing e.g. `--lib` or `--bin NAME` to specify a single target
```

I guess that's definitely a problem with flycheck-rust.

Thank you so much for your help.

**Comment by liuchong** [comment]
so, is is solved? I have the same problem.

**Comment by fbergroth** [comment]
@liuchong `rustfmt` is bundled with `rust-mode` now, use that instead of this package.

**Comment by liuchong** [comment]
@fbergroth Thanks, I'm using a "spacemacs" with default rust configuration (just clone the .emacs.d and then add a rust layer, nothing else...).
In fact there is no such problem when I use my own configuration without spacemacs. not sure what wrong of it, but I think spacemacs is using rust-mode.

**Comment by liuchong** [comment]
https://github.com/syl20bnr/spacemacs/tree/master/layers/%2Blang/rust
This is the link, and the parentheses turn red after I type C-c C-f

**Comment by fbergroth** [comment]
I wrote some of the rust layer in spacemacs so I'm familiar with it.

Anyway, the function you're calling is defined in `rust-mode` and not from this obsolete package, so I suggest you post the issue there.

**Comment by liuchong** [comment]
You mean open another ticket to describe the issue?
but what I've seen is that the parentheses turn red after code formated, don't know how to address the problem.
would you have same issue in spacemacs?
I can re-run a M-x rust-mode(or any other mode) and then the red parentheses disappeared, so seems like that some plugins of spacemacs caused "...a missing of 'refersh'..."

**Comment by liuchong** [comment]
oh, sorry red "there" as "here"... let me make a issue in the rust-mode :new_moon_with_face:
